### PR TITLE
Update docs and generate id script

### LINKED ISSUE #148: Move hamburger menu to the top left
As per image in issue #147, the menu button is on the bottom and is odd

### LINKED ISSUE #149: Update Web Docs to match Readme
GitHub readme shows Shabad OS logo, web docs has a broken link to schema.

### LINKED ISSUE #1573: add cli count and type parameter for generate-id script
script runs slowly, would help to output multiple lines (e.g. `npm run generate-id 30`) and to not waste energy outputting both shabad and line ids when only one is required (e.g. `npm run generate-id -count 30 -no-shabad-id`)

### PR DESCRIPTION
**Pull request by saihaj**
For #147 cannot seem to find out why the homepage is not auto sizing itself because we have meta tags for HTML 5.
Also updates the `lib/generate-id` 

Closes #148
Closes #149 
Closes #1573

### CODE REVIEW

File: lib/generate-id.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
+const commander = require( 'commander' )
```
Comment (Harjot1Singh) [inline lib/generate-id.js:1]: Minor style nit, but newline after global modules vs local imports

File: lib/generate-id.js [hunk 2]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
 }
 
 /**
- * Generates a line id and a shabad id, using the database.
+   * Logs the generated id to the console
+   * @param {Number} count the count of id to generate
+   * @param {String} type shabad, line
+   */
+const logGeneratedID = async function ( count, type ) {
```
Comment (Harjot1Singh) [inline lib/generate-id.js:29]: You've repeated yourself a little for each time: a better approach would be to store all types in an object with the corresponding function. E.g:

```
const idTypes = {
  shabad: () => findId( Shabads, 3 )
  line: () => findId( Lines, 4 )
}

// your function

// Retrieve the correct generator function for our type
const generateId = idTypes[ type ]

// only need one loop now
console.log( generateId() )
```

This way, it's extremely simple and maintainable.

File: lib/generate-id.js [hunk 3]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
...
+  let i = 0
```
Comment (Harjot1Singh) [inline lib/generate-id.js:30]: In JS, put your let before the variable in the loop. It is also preferable to use `i++` (but 90% of the time, we can avoid `for` loops altogether in favour of `.map` etc, this is one of the reasonable exceptions)

```
for ( let i = 0; i < count; i++ )
```

File: lib/generate-id.js [hunk 4]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
...
+
+  if ( type === 'shabad' ) {
+    for ( i = 0; i < count; i += 1 ) {
+      console.log( `${i + 1}. New Shabad ID: `.bold.magenta + `${await findId( Shabads, 3 )}`.rainbow )
+    }
+  } else if ( type === 'line' ) {
+    for ( i = 0; i < count; i += 1 ) {
+      console.log( `${i + 1}. New Line ID: `.bold.magenta + `${await findId( Lines, 4 )}`.rainbow )
+    }
+  } else { // Default log all types
```
Comment (Harjot1Singh) [inline lib/generate-id.js:40]: I'd say the default behaviour should definitely be to require a type, I can't think of a scenario where you'd want to generate the same number of IDs for multiple types.

File: lib/generate-id.js [hunk 5]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
...
+   * Logs the generated id to the console
+   * @param {Number} count the count of id to generate
+   * @param {String} type shabad, line
+   */
+const logGeneratedID = async function ( count, type ) {
+  let i = 0
+
+  if ( type === 'shabad' ) {
+    for ( i = 0; i < count; i += 1 ) {
+      console.log( `${i + 1}. New Shabad ID: `.bold.magenta + `${await findId( Shabads, 3 )}`.rainbow )
```
Comment (Harjot1Singh) [inline lib/generate-id.js:34]: I know the old behaviour had "New Shabad ID" before each, but I was thinking it might be better to scrap this and just output the ID, purely because you may wish to do:

`npm run generate-id -- -t line -c 100 > line_ids.txt`

and then you've got a machine-parsable file by default.

File: lib/generate-id.js [hunk 6]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
...
+
+/**
+ * Interpret user command and print the id, using the database.
  */
 const main = async () => {
-  console.log( 'New Line ID: '.bold.cyan + `${await findId( Lines, 4 )}`.rainbow )
-  console.log( 'New Shabad ID: '.bold.magenta + `${await findId( Shabads, 3 )}`.rainbow )
+  commander
+    .version( '0.0.1' )
+    .description( 'Generate ID\'s for ShabadOS Database \n\nCommand \nshabad \t\tgenerate id\'s for shabads \nline \t\tgenerate id\'s for line' )
```
Comment (Harjot1Singh) [inline lib/generate-id.js:54]: A hint: you can use backticks ``` to signify multiline strings, or strings you need to escape, making it a little more natural in the code. I'd also use the the `idTypes` object mentioned above to generate a string list of things you can generate, e.g.:

`Object.keys( idTypes ).join(', ')`

File: lib/generate-id.js [hunk 7]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
...
+    .option( '-c, --count <integer>', 'number of ID\'s', 1 )
+    .option( '-t, --type [command]', 'generate ID for only [command]' )
```
Comment (Harjot1Singh) [inline lib/generate-id.js:56]: Rename `[command]` to `[type]` to avoid confusion

Comment (Harjot1Singh) [reply lib/generate-id.js:56]: Or is there a specific reason you used `[command]`?

File: lib/generate-id.js [hunk 8]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
...
+    .parse( process.argv )
+
+  await logGeneratedID( commander.count, commander.type )
```
Comment (Harjot1Singh) [inline lib/generate-id.js:59]: A tiny nitpick, but here it makes to destructure the the commander parameters and pass them in

```
const { count, type } = commander

// It's easier in the future to switch out the source of these parameters because of the destructure
await logGeneratedID( count, type)
```

File: lib/generate-id.js (hunk 2 above)
Comment (Harjot1Singh) [inline lib/generate-id.js:29]: Minor nitpick about the function declaration:
1) It logs multiple IDs, so probably better to use the plural here
2) Keep `ID` as `Id`/follow the camelCase rules for abbreviations too
3) Don't use `function` unless absolutely necessary. It has slightly different semantics to the arrow declarations: `const logGeneratedId = async (count, type) => {`
4) You're using the past-tense in the function name, which would indicate that the IDs are already generated when you call the function. I'd personally rename it to `generateIds`.

File: lib/generate-id.js [hunk 9]
```diff
@@ -22,11 +22,41 @@ const findId = async ( model, length ) => {
 }
 
 /**
- * Generates a line id and a shabad id, using the database.
+   * Logs the generated id to the console
```
Comment (Harjot1Singh) [inline lib/generate-id.js:25]: The JSDoc is out of line, these stars should only be 1 space indented in

File: lib/generate-id.js (hunk 5 above)
Comment (Harjot1Singh) [reply lib/generate-id.js:34]: Performance issue: Each time `findId` is called, it queries the database and generates a set of IDs that are taken. You should refactor this to have the the DB only queried once.

Logic bug: New IDs that are generated are not filtered out, and therefore generated new IDs can have duplicates. You need to update the set, so maybe move the function and set into this function.

File: docs/usage/overview.md [hunk 10]
```diff
@@ -3,10 +3,11 @@
 This section outlines methods of using of the database, and goes through some examples.
 
 There are several methods of using the database, depending on your purpose:
-- [**General Browsing/Querying**](#general-browsingquerying): Use an application such as [DBeaver](https://dbeaver.io/) to query the  [SQLite database](https://github.com/ShabadOS/database/releases)
-- [**Offline App**](#offline-applications-sqlite): Download the [SQLite database](https://github.com/ShabadOS/database/releases)
-- [**Offline JavaScript App (recommended)**](#offline-applications-javascript): Install the [API via npm](installation#npm)
-- [**Online App**](#online-applications-gurbaninow-api): Use the [GurbaniNow API](https://github.com/GurbaniNow/api)
+- [Overview](#overview)
+  - [General Browsing/Querying](#general-browsingquerying)
```
Comment (Harjot1Singh) [inline docs/usage/overview.md:7]: Btw I'd always put commits like this that are unrelated in a separate PR (branched cleanly off `dev`) so that if there's a disagreement on the unrelated changes, the PR doesn't get stuck.

File: lib/generate-id.js (hunk 3 above)
Comment (saihaj) [reply lib/generate-id.js:30]: If I use `i++` then eslint gives me following error ` Unary operator '++' used.eslintno-plusplus `

File: lib/generate-id.js (hunk 6 above)
Comment (saihaj) [reply lib/generate-id.js:54]: eslint: Multiline support is limited to browsers supporting ES5 only `no-multi-str`

File: lib/generate-id.js [hunk 11]
```diff
@@ -21,12 +20,47 @@ const findId = async ( model, length ) => {
...
+
 /**
- * Generates a line id and a shabad id, using the database.
- */
+  * Logs the id to the console
+  * @param {Number} count the count of id to generate
+  * @param {String} type check `availableTypes`
+  */
+const logGenerateIds = async ( count, type ) => {
+  const newId = idTypes[ type ]
```
Comment (Harjot1Singh) [inline lib/generate-id.js:34]: The name here confused me a little - something like `idType` or `idGenerator` would be better imo

File: lib/generate-id.js [hunk 12]
```diff
@@ -21,12 +20,47 @@ const findId = async ( model, length ) => {
...
+
+  // Grab ids to exclude from database
+  const ids = new Set( ( await newId[ 0 ].query().select( 'id' ) ).map( ( { id } ) => id ) )
+
+  // Loop to print id's
+  for ( let i = 0; i < count; i += 1 ) {
+    console.log( `${await findId( ids, newId[ 1 ] )}` )
```
Comment (Harjot1Singh) [inline lib/generate-id.js:41]: `newId[ 1 ] ` is a bit obfuscated in meaning. I can tell you're passing in the length of the ID from reading all the code, but not from solely scanning that line.

Instead, destructure it before the loop (along with line 37's use of `newId[ 0 ]`):

`const [ Model, idLength ] = newId`

Comment (Harjot1Singh) [inline lib/generate-id.js:41]: YOu also need to update the set with the new ID when you find it, otherwise you may end up generating duplicates, so somewhere `ids.add( newId )`

File: lib/generate-id.js [hunk 13]
```diff
@@ -21,12 +20,50 @@ const findId = async ( model, length ) => {
...
+    .version( '0.0.1' )
+    .description( "Generate ID\'s for ShabadOS Database \n\n Type\n shabad \t\tgenerate id\'s for shabads \nline \t\tgenerate id\'s for line" )
+    .option( '-c, --count <integer>', 'number of ID\'s', 1 )
+    .requiredOption( '-t, --type [type]', 'generate ID for only [type]' )
+    .parse( process.argv )
+
+  const { count, type } = commander
+  try {
+    await logGenerateIds( count, type )
+  } catch ( e ) {
```
Comment (Harjot1Singh) [inline lib/generate-id.js:62]: What if there is another type of error?

Comment (Harjot1Singh) [reply lib/generate-id.js:62]: Currently, this will suppress all other errors mistakenly.

https://github.com/tj/commander.js#custom-option-processing on how to supply a 4th parameter which is a custom validator/processor, throw the error there, and then `console.error(err.message)` once here

File: lib/generate-id.js [hunk 14]
```diff
@@ -21,12 +20,50 @@ const findId = async ( model, length ) => {
...
+
 /**
- * Generates a line id and a shabad id, using the database.
- */
+  * Logs the id to the console
+  * @param {Number} count the count of id to generate
+  * @param {String} type check `availableTypes`
+  */
+const logGenerateIds = async ( count, type ) => {
+  const idType = idTypes[ type ]
```
Comment (Harjot1Singh) [inline lib/generate-id.js:34]: Even better, combine the lines

File: lib/generate-id.js [hunk 15]
```diff
@@ -1,17 +1,16 @@
-const { generateId } = require( './utils' )
-const colors = require( './string-colors' )
+const commander = require( 'commander' )
 
+const { generateId } = require( './utils' )
 const { Shabads, Lines } = require( '../' )
 
+const availableTypes = [ 'shabad', 'line' ]
```
Comment (Harjot1Singh) [inline lib/generate-id.js:6]: This is actually a duplicate of `idTypes`, since `Object.keys(idTypes)` will give you the exact same, and you don't want to be maintaining two lists if you can help it. So just use the aforementioned where you need it, since I can see 1 place, and remove this var

File: lib/generate-id.js (hunk 13 above)
Comment (saihaj) [reply lib/generate-id.js:62]: so if it is something other than invalid type error then it should throw the `console.error(err.message)` otherwise do what I already have there? right?

Comment (saihaj) [reply lib/generate-id.js:62]: > What if there is another type of error?

I can have a conditional catch-block

```
try {
    await logGenerateIds( count, type )
  } catch ( e ) {
    if ( e instanceof TypeError ) {
      console.log( 'Invalid value for type parameter\n' )
      console.log( 'Available types are: \n' )
      console.log( Object.keys( idTypes ).join( '\n' ) )
    } else {
      console.error( e )
    }
  }
```
